clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3091,6 +3104,89 @@ static void handleMaxGlobalWorkDimAttr(Sema &S, Decl *D,
                                                                      E);
 }
 
+SYCLIntelLoopFuseAttr *
+Sema::mergeSYCLIntelLoopFuseAttr(Decl *D, const AttributeCommonInfo &CI,
+                                 Expr *E) {
+
+  if (const auto ExistingAttr = D->getAttr<SYCLIntelLoopFuseAttr>()) {
+    // [[intel::loop_fuse]] and [[intel::loop_fuse_independent]] are
+    // incompatible
+    if (ExistingAttr->getAttributeSpellingListIndex() !=
```

Comment:
Yea I thought about this a while. I couldn't  think of another way as well, other than a bunch of if else checks. If a GNU (or other spelling) is added, I think we'll need to define more accessors and maybe member functions?

Reply:
The problem is -- the accessors are generated for the semantic attribute, so they're basically useless for distinguishing between different parse forms.

I think for now, just add a comment that says something along the lines of "FIXME: if additional spellings are provided for the attributes, this code will silently do the wrong thing."

Reply:
>>The problem is -- the accessors are generated for the semantic attribute, so they're basically useless for distinguishing between different parse forms.

Would you mind explaining this? I just took a look at the .inc file and see this generated for the isIndependent accessor - 

`bool isIndependent() const { return getAttributeSpellingListIndex() == 1; }`

Maybe I'm misunderstanding something but I don't understand why this cannot be used to distinguish between the various spellings.

Reply:
Happy to explain!

We have attributes at two different stages of the compiler. We have parsed attributes that are generated by the parser and we have semantic attributes that are AST nodes. The code in SemaDeclAttr.cpp is what maps parsed attributes onto semantic attributes.

Parsed attributes are pretty stupid in terms of capabilities -- they keep track of things like what spelling was used, what arguments were passed to the attribute, what syntax was used, etc. It's all very generic in that it can represent all the parsed data for any kind of attribute. Semantic attributes are smarter in that ClangAttrEmitter.cpp generates a bunch of code for us based on what's in Attr.td. The basic idea is: each attribute defined in Attr.td is a single semantic attribute and each of the spellings listed in Attr.td are the valid ways you can get a parsed attribute that will map to the semantic attribute.

Accessors are a thing for semantic attributes -- the code in ClangAtterEmitter.cpp emits the accessors for you when generating the AST node for the attribute. That's why you see it in the .inc file. But nothing is generated for the parsed attribute interface, so you can't do something like:
```
void handleLoopFuseAttr(Sema &S, Decl *D, const ParsedAttr &Attr) {
  if (Attr.isIndependent()) { ... }
}
```
Does that make more sense?

Reply:
Yes that does. Thanks for explaining!

Reply:
Added the comment.